src/rstar.jl

```diff
@@ -0,0 +1,91 @@
+"""
+    rstar(chains::Chains, model::MLJModelInterface.Supervised; subset = 0.8, iterations = 10, verbosity = 0)
+    rstar(rng::Random.AbstractRNG, chains::Chains, model::MLJModelInterface.Supervised; subset = 0.8, iterations = 10, verbosity = 0)
+    rstar(rng::Random.AbstractRNG, model::MLJModelInterface.Supervised, x::AbstractMatrix, y::AbstractVector; subset = 0.8, iterations = 10, verbosity = 0)
+
+Compute the R* statistic for convergence diagnostic of MCMC. This implementation is an adaption of Algorithm 1 & 2, described in [Lambert & Vehtari]. Note that the correctness of the statistic depends on the convergence of the classifier used internally in the statistic. You can track if the training of the classifier converged by inspection of the printed RMSE values from the XGBoost backend. To adjust the number of iterations used to train the classifier set `niter` accordingly.
```

Comment:
It seems a bit strange that the classifier has to be specified first when provide data as matrix and vector, but after the chains argument in the other case. Might be good to decide for one consistent form. Also might be useful to allow to call `rstar` with matrix+vector argument without specifying an RNG.

Maybe also simplify the docstring a bit according to the [docs](https://docs.julialang.org/en/v1/manual/documentation/#Documentation), e.g., it is not necessary to specify the keyword arguments in the signature since you list them in a separate section. I think it's also not needed to add all type annotations (in particular for the RNG). E.g., something like:
```suggestion
    rstar([rng ,]model, chains::Chains; kwargs...)
    rstar([rng ,]model, x::AbstractMatrix, y::AbstractVector; kwargs...)

Compute the R* convergence diagnostic of MCMC.

This implementation is an adaption of Algorithm 1 & 2, described in [Lambert & Vehtari]. Note that the correctness of the statistic depends on the convergence of the classifier used internally in the statistic. You can track if the training of the classifier converged by inspection of the printed RMSE values from the XGBoost backend. To adjust the number of iterations used to train the classifier set `niter` accordingly.
```
Maybe even more descriptive to use `classifier` instead of `model` in the docs?

Reply:
Yes, good points. Thanks.

Reply:
I reopened this conversation since it seems the latest commits did not address the comments about the method inconsistency and the version without RNG.

Reply:
It’s already resolved locally but I didn’t have time to push. Will do tomorrow.